src/util/sockopt.rs

```diff
@@ -27,12 +27,11 @@ mod unix {
     use std::io::{Result, Error};
     use std::os::unix::io::AsRawFd;
 
-    use mio::tcp::TcpStream;
     use nix::Error as NixError;
     use nix::sys::socket;
     use nix::sys::socket::sockopt;
 
-    impl SocketOpt for TcpStream {
+    impl<T: AsRawFd> SocketOpt for T {
```

Comment:
the function for handing socket is set_socket_opt or get_socket_opt, I think ok.